Added options parameter to showChildView()

jbarreiros opened the pull request:
Fixes #2572.

> ... `showChildView` does not take an `options` hash.

This PR changes `showChildView(view)` to `showChildView(view, options)` to provide consistency with `Marionette.Region.show(view, options)`.

I'm new to writing unit tests, so please let me know if the unit test needs adjusting.

paulfalgout commented:
This likely needs a little bit of modification to documentation

samccone commented on src/view.js:
```diff
@@ -144,8 +144,8 @@ Marionette.View = Marionette.AbstractView.extend({
-  showChildView: function(regionName, view) {
-    return this.getRegion(regionName).show(view);
+  showChildView: function(regionName, view, options) {
+    return this.getRegion(regionName).show(view, options);
```
lets do 

```
this.getRegion(regionName).show.apply(this, _.rest(arguments)
```

jasonLaster replied:
yep, @jbarreiros the benefit is that we can mantain the airity of the options

jbarreiros replied:
Good idea. Updated.

samccone commented:
@jbarreiros mind adding docs and we can :ship: this

jbarreiros commented:
@samccone Looking for a little direction on where to update the documentation. Do we just want to highlight that `options` can be passed to both `Region.show` and `Region.showChildView`?

samccone commented:
that would be perfect @jbarreiros

jasonLaster commented:
ping @jbarreiros - have a minute to update the docs to include `options`?

jbarreiros commented:
Added the updates to the docs. Sorry about the extreme delay.

In writing the docs and reading through the code, I noticed that `options` is [not passed to `empty()`](https://github.com/marionettejs/backbone.marionette/blob/next/src/region.js#L74). This actually contradicts the docs: 
![selection_002](https://cloud.githubusercontent.com/assets/1109194/8867156/cc27f88a-3193-11e5-8d73-d317c76cba52.png)

I cleaned up the line mentioning "you also have access to the `options` ...".

ahumphreys87 commented:
lgtm :+1: @samccone @jasonLaster  could we get another set of :eyes:
